### PR TITLE
[syncd swap] Added MASIC platform support for swapping syncd with syncd-rpc

### PR DESCRIPTION
**Pull request by smaheshm**
### Description of PR

Summary:
Added MASIC platform support for swapping syncd with syncd-rpc.

On multi ASIC platforms calling swap syncd API will swap syncd's on all namespaces. There is no option to update syncd for a specific namespace.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Add support for multi ASIC platform.

#### How did you do it?
Updated APIs to iterate over asics in case of Multi ASIC platform.

#### How did you verify/test it?

Ran 'swap_syncd' API on single and multi ASIC platform.

```
samaddik@str-serv-acs-14:/var/sonic-mgmt/tests$ pytest dummy/test_params_dummy2.py --testbed vms12-t1-lag-1 --inventory=../ansible/str,../ansible/veos,../ansible/str2,../ansible/strsvc --testbed_file=../ansible/testbed.csv --host-pattern=str-s6000-acs-8  --module-path=../ansible/library --disable_loganalyzer --skip_sanity 
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
================================================================================ test session starts =================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /var/sonic-mgmt/tests, inifile: pytest.ini
plugins: forked-1.3.0, metadata-1.11.0, xdist-1.28.0, html-1.22.1, repeat-0.9.1, ansible-2.2.2
collected 1 item                                                                                                                                                                     

dummy/test_params_dummy2.py .                                                                                                                                                  [100%]

============================================================================= 1 passed in 257.79 seconds =============================================================================
samaddik@str-serv-acs-14:/var/sonic-mgmt/tests$ 
samaddik@str-serv-acs-14:/var/sonic-mgmt/tests$ pytest dummy/test_params_dummy2.py --testbed vms12-t1-lag-1 --inventory=../ansible/str,../ansible/veos,../ansible/str2,../ansible/strsvc --testbed_file=../ansible/testbed.csv --host-pattern=str-s6000-acs-8  --module-path=../ansible/library --disable_loganalyzer --skip_sanity 
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
================================================================================ test session starts =================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /var/sonic-mgmt/tests, inifile: pytest.ini
plugins: forked-1.3.0, metadata-1.11.0, xdist-1.28.0, html-1.22.1, repeat-0.9.1, ansible-2.2.2
collected 1 item                                                                                                                                                                     

dummy/test_params_dummy2.py .                                                                                                                                                  [100%]

============================================================================= 1 passed in 260.45 seconds =============================================================================
```

------MASIC------

```
samaddik@svcstr-server-2:/var/sonic-mgmt/tests$  pytest dummy/test_params_dummy2.py  --testbed=vmsvc1-t1-masic-acs-1 --inventory=../ansible/strsvc  --testbed_file=../ansible/testbed.csv --host-pattern=svcstr-masic-acs-1  --module-path=../ansible/library --disable_loganalyzer --skip_sanity 
/usr/local/lib/python2.7/dist-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.exceptions import InvalidSignature
================================================================================ test session starts =================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /var/sonic-mgmt-w/tests, inifile: pytest.ini
plugins: html-1.22.1, repeat-0.9.1, forked-1.3.0, xdist-1.28.0, metadata-1.11.0, ansible-2.2.2
collected 4 items                                                                                                                                                                    

dummy/test_params_dummy2.py ....                                                                                                                                               [100%]

============================================================================ 4 passed in 1156.38 seconds =============================================================================
```

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -1850,6 +1953,8 @@ class MultiAsicSonicHost(object):
     """
 
     _DEFAULT_SERVICES = ["pmon", "snmp", "lldp", "database"]
+    MULTI_ASIC_SERVICE_NAME = "{}@{}"   # service name, asic_id
+    MULTI_ASIC_DOCKER_NAME = "{}{}"     # docker name,  asic_id
```
Comment (smaheshm) [inline tests/common/devices.py:1957]: Delete.

File: tests/common/devices.py [hunk 2]
```diff
@@ -390,6 +390,23 @@ def is_service_fully_started(self, service):
...
+        Returns:
+            True or False
+        """
+        status = self.command(
+            "docker ps -f name={}".format(service),
+            module_ignore_errors=True
+        )["stdout_lines"]
+        if len(status) > 1:
+            return True
+        return False
```
Comment (daall) [inline tests/common/devices.py:408]:
```suggestion
        return len(status) > 1
```

File: tests/common/devices.py [hunk 3]
```diff
@@ -1262,6 +1279,48 @@ def get_crm_facts(self):
...
+            ),
+            module_ignore_errors=True
+        )["stdout"]
+
+        logging.info(
+            "{} status:\n%s".format(docker_name),
+            self.command(
+                "docker ps -f name={}".format(docker_name), module_ignore_errors=True
+            )["stdout"]
+        )
```
Comment (daall) [inline tests/common/devices.py:1320]: Would it make more sense to move this log up to the `is_container_present` call and add a log here to show the `supervisorctl` output?

Comment (smaheshm) [reply tests/common/devices.py:1320]: Sounds fine. I added it here since in your latest commit this was log present after checking the service liveness.

**Review by daall — APPROVED** [review]

File: tests/common/devices.py (hunk 3 above)
Comment (daall) [reply tests/common/devices.py]: Wait, I see what you're saying now. I think moving the `docker ps -f` print above is still a good change, but we should still have a log message for `docker exec ... supervisorctl status` here.

Comment (smaheshm) [reply tests/common/devices.py]: sure, I'll add.

**Review by daall — APPROVED** [review]

**Pull request merged**